pkg/api/validation/validation.go

```diff
@@ -649,3 +649,29 @@ func ValidateResourceName(str string) errs.ValidationErrorList {
 
 	return errs.ValidationErrorList{}
 }
+
+// ValidateLimitRange tests if required fields in the LimitRange are set.
+func ValidateLimitRange(limitRange *api.LimitRange) errs.ValidationErrorList {
+	allErrs := errs.ValidationErrorList{}
+	if len(limitRange.Name) == 0 {
+		allErrs = append(allErrs, errs.NewFieldRequired("name", limitRange.Name))
+	} else if !util.IsDNSSubdomain(limitRange.Name) {
+		allErrs = append(allErrs, errs.NewFieldInvalid("name", limitRange.Name, ""))
+	}
+	if len(limitRange.Namespace) == 0 {
+		allErrs = append(allErrs, errs.NewFieldRequired("namespace", limitRange.Namespace))
+	} else if !util.IsDNSSubdomain(limitRange.Namespace) {
+		allErrs = append(allErrs, errs.NewFieldInvalid("namespace", limitRange.Namespace, ""))
+	}
+	// ensure resource names are properly qualified per docs/resources.md
+	for i := range limitRange.Spec.Limits {
+		limit := limitRange.Spec.Limits[i]
+		for k := range limit.Max {
+			allErrs = append(allErrs, ValidateResourceName(string(k))...)
+		}
+		for k := range limit.Min {
+			allErrs = append(allErrs, ValidateResourceName(string(k))...)
+		}
```

Comment:
Do you need to reject resource names other than cpu or ram, given that your implementation silently ignores other resource types?